pkg/provider/gke/gke.go

```diff
@@ -43,6 +43,10 @@ import (
 
 // New is the GKE constructor.
 func New(dr *provider.DeploymentResource) *GKE {
+
+	// Set SERVICEACCOUNT_CLIENT_EMAIL to LoadBalancer.
+	dr.FlagDeploymentVars["NGINX_SERVICE_TYPE"] = "LoadBalancer"
```

Comment:
We have a new function to set this now. See `SetupDeploymentResources` from #413